CMakeLists.txt

```diff
@@ -110,3 +124,7 @@ if(MM_USE_EXTRA_WARNINGS)
 endif()
 
 add_subdirectory(src)
+
+if(MM_BUILD_TEST)
+    add_subdirectory(test)
+endif()
```

Comment:
Is there a way to build the tests _after_ the `src` files, so that if we break something, it fails fast and does not even attempt to build these tests?

I could see this being an automatic feature; so if this is already done, please just add this as a comment.

Reply:
I think we would have to build our common stuff as a lib and link it in test for that to happen.